webapp/channels/src/packages/mattermost-redux/src/selectors/entities/users.ts

```diff
@@ -223,6 +223,27 @@ export const getCurrentUserMentionKeys: (state: GlobalState) => UserMentionKey[]
     },
 );
 
+export type UserHighlightKey = {
+    key: string;
+    caseSensitive?: boolean;
```

Comment:
Does having this field make using the existing code in `text_formatting.ts` easier, or can we leave it out? I can't imagine people will ever want this to be case sensitive, but there's one case that we need case sensitive mentions (just for first names and just because some people will be named words like Will that overlap with common words)

Reply:
removed case sensitive key